### PR TITLE
[sdk] Update to Gecko SDK 4.2.1

### PR DESCRIPTION
**Pull request by lmnotran**
This updates the gecko_sdk submodule to 4.2.1 and also updates the PAL in src/src

# Known Issues
- `crypto.c` changes were not pulled into the ot-efr32 PAL in `src/src`. See the Known Issues of https://github.com/openthread/ot-efr32/pull/482 for more details

### CODE REVIEW

File: src/src/sl_rcp_gp_interface.c [hunk 1]
```diff
@@ -1,55 +1,43 @@
...
- *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- *  POSSIBILITY OF SUCH DAMAGE.
- */
-
-/*******************************************************************************
- * @file
- * @brief This file implements Green Power interface.
- ******************************************************************************/
+/***************************************************************************/ /**
```
Comment (hemanth-silabs) [inline src/src/sl_rcp_gp_interface.c:1]: @lmnotran make-pretty issue?

File: src/src/sl_rcp_gp_interface.c [hunk 2]
```diff
@@ -132,51 +127,58 @@ void efr32GpProcess(void)
...
 {
     bool     isGpPkt           = false;
     uint8_t *gpFrameStartIndex = efr32GetPayload(aFrame);
-    otLogDebgPlat("GP RCP INTF: PL Index = %d", (gpFrameStartIndex - aFrame->mPsdu));
 
     // A Typical MAC Frame with GP NWK Frame in it
-    /* clang-format off */
-    // MAC Frame  : [<---------------MAC Header------------->||<------------------------------------NWK Frame----------------------------------->]
-    //               FC(2) | Seq(1) | DstPan(2) | DstAddr(2) || FC(1) | ExtFC(0/1) | SrcId(0/4) | SecFc(0/4) | MIC(0/4) | <------GPDF(1/n)------>
+    // MAC Frame  : [<---------------MAC Header------------->||<------------------------------------NWK
```
Comment (hemanth-silabs) [inline src/src/sl_rcp_gp_interface.c:142]: Same here.

File: src/src/sl_rcp_gp_interface.h [hunk 3]
```diff
@@ -1,50 +1,36 @@
...
- *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- *  POSSIBILITY OF SUCH DAMAGE.
- */
-
-/*******************************************************************************
- * @file
- * @brief This file implements Green Power interface.
- ******************************************************************************/
+/***************************************************************************/ /**
```
Comment (hemanth-silabs) [inline src/src/sl_rcp_gp_interface.h:1]: Same

**Review by hemanth-silabs — CHANGES_REQUESTED** [review]
@lmnotran seems like there are some make-pretty issues.

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by parag-silabs — APPROVED** [review]
LGTM

**Review by bertoldi-silabs — APPROVED** [review]

**Review by hemanth-silabs — APPROVED** [review]
👍

**Comment by lmnotran** [comment]
@jwhui can you review this please?

**Review by jwhui — APPROVED** [review]

**Comment by jwhui** [comment]
@lmnotran , can you help rebase and resolve conflicts?

**Comment by lmnotran** [comment]
@jwhui Thanks for taking a look at the PR! You should be good to merge now

**Review by jwhui — APPROVED** [review]

**Pull request merged**